### PR TITLE
[Misc] Add submodule Eigen 

### PR DESCRIPTION
**Pull request by FantasyVR**
Related pr = https://github.com/taichi-dev/taichi/pull/2657

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -37,3 +37,7 @@
 [submodule "external/glm"]
 	path = external/glm
 	url = https://github.com/g-truc/glm.git
+[submodule "external/eigen"]
+	path = external/eigen
+	url = https://gitlab.com/libeigen/eigen.git
+	branch = 3.3
```
Comment (yuanming-hu) [inline .gitmodules:43]: Thanks! Would it be possible to use tag [3.3.9](https://gitlab.com/libeigen/eigen/-/tree/3.3.9) or simply commit 0fd6b4f7?

**Review by k-ye — APPROVED** [review]
LGTM!

File: .gitmodules (hunk 1 above)
Comment (FantasyVR) [reply .gitmodules:43]: Fix it. 
BTW, the git submodule doesn't support update the submodule by tag.  Although the branch you saw is 3.3, I checkout tag 3.3.9 (commit 0fd6b4f7) locally and then recommit. Now, the submodule actually is tag 3.3.9.

**Pull request merged**